### PR TITLE
Search name city and update location on map

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Feature 73915: Search screen](https://edu-redmine.sun-asterisk.vn/issues/73915)
## WHAT
- Create UI tableView search screen
- Handle action: search name city and request response from Apple Map
- Handle action: Update data for tableview
- Handle action: select row of location and update location on map
## Evidence (Screenshot or Video)
https://github.com/awesome-academy/HCM_iOS_2024_Weather_App/assets/161679147/1a64c3ff-3204-4a45-b5db-183f1d08ea91
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: Weather_App/Weather_App/Scenes/Map/LocationSearch/LocationSearchController.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        tableView.register(LocationCell.self, forCellReuseIdentifier: "LocationCell")
+    }
+    
+}
+
+extension LocationSearchController: UISearchResultsUpdating {
+    internal func updateSearchResults(for searchController: UISearchController) {
+        guard let searchText = searchController.searchBar.text, !searchText.isEmpty else {
+            searchResults.removeAll()
+            tableView.reloadData()
```
Comment (dongtg-1485) [inline Weather_App/Weather_App/Scenes/Map/LocationSearch/LocationSearchController.swift]: `tableView.reloadData()` gọi luôn trong didSet của `searchResults`

File: Weather_App/Weather_App/Scenes/Map/LocationSearch/LocationSearchController.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            return
+        }
+        getSearchResults(for: searchText)
+    }
+    
+    private func getSearchResults(for searchText: String) {
+        let request = MKLocalSearch.Request()
+        request.naturalLanguageQuery = searchText
+        let search = MKLocalSearch(request: request)
+        search.start { [weak self] responseSearch, error in
```
Comment (dongtg-1485) [inline Weather_App/Weather_App/Scenes/Map/LocationSearch/LocationSearchController.swift]: logic search để vào trong Repository, ViewController chỉ nhận gửi input và nhận output

File: Weather_App/Weather_App/Scenes/Map/LocationSearch/LocationSearchController.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+extension LocationSearchController {
+    override internal func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchResults.count
+    }
+    
+    override internal func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "LocationCell", for: indexPath) as! LocationCell
+        let mapItem = searchResults[indexPath.row].placemark
+        cell.textLabel?.text = mapItem.name
```
Comment (dongtg-1485) [inline Weather_App/Weather_App/Scenes/Map/LocationSearch/LocationSearchController.swift]: tạo function `setContent()` thay vì chọc thằng vào properties

**Review by daolq-2712 — APPROVED** [review]
Ok!

**Pull request merged**